src/hash_generator/EPCISEventToPreHashedString.js

```diff
@@ -0,0 +1,449 @@
+ * })
+ * @param {boolean} throwError - if set to true, it will throw an error if the event misses some
+ * fields for example. Otherwise, it won't throw an error and it will still return the generated id
+ * @returns {string} - the pre-hashed string
+ */
+export const getPreHashStringOfElementWithChildren = (element, context, throwError) => {
+  let string = '';
+  const strings = [];
+
+  Object.keys(element).forEach((key) => {
+    strings.push(getPreHashStringFromCustomFieldElement(key, element[key], context, throwError));
+  });
+
+  string += listOfStringToPreHashLexicalOrderedString(strings);
+
+  return string;
+};
+
+/**
+ * Return the pre-hashed string corresponding to the list passed in parameter + the pre-hashed
+ * strings of custom fields
+ * @param {[]} list - the list to convert
+ * @param {{}} context - the list of context (e.g {
+ *    "example": "http://ns.example.com/epcis/",
+ *    "example2": "http://ns.example2.com/epcis/",
+ * })
+ * @param {string} fieldName - the field name of the list
+ * @param {boolean} throwError - if set to true, it will throw an error if the event misses some
+ * fields for example. Otherwise, it won't throw an error and it will still return the generated id
+ * @returns {*}
+ */
+export const preHashStringTheList = (list, context, fieldName, throwError) => {
+  let string = fieldName;
+  const strings = [];
+  const customFields = [];
+  let res;
+
+  if (list.length === 0) return ''; // rule n°4
+
+  switch (fieldName) {
+    case 'epcList':
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('epc', list[i], throwError));
+      }
+      break;
+    case 'childEPCs':
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('epc', list[i], throwError));
+      }
+      break;
+    case 'outputEPCList':
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('epc', list[i], throwError));
+      }
+      break;
+    case 'inputEPCList':
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('epc', list[i], throwError));
+      }
+      break;
+    case 'set':
+      string = '';
+      // todo: see if we need to replace 'completeness_inferred'
+      // by 'urn:epcglobal:cbv:disp:completeness_inferred' for example
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('set', list[i], throwError));
+      }
+      break;
+    case 'unset':
+      string = '';
+      // todo: see if we need to replace 'completeness_inferred'
+      // by 'urn:epcglobal:cbv:disp:completeness_inferred' for example
+      for (let i = 0; i < list.length; i += 1) {
+        strings.push(getPreHashStringOfField('unset', list[i], throwError));
+      }
+      break;
+    case 'quantityList':
+      for (let i = 0; i < list.length; i += 1) {
+        res = getPreHashStringOfElementAccordingToOrderList(list[i], context,
```

Comment:
More succinct function and variable names would make this `switch` easier to follow:
```js
res = getOrderedPreHashString(list[i], context, quantityElementOrder, throwError);
```